client/src/common/helpers.js

```diff
@@ -1,266 +0,0 @@
-
-    return !headers
-      ? asyncFetch(fullUrl)
-      : asyncFetch(fullUrl, {
-          method: 'GET',
-          headers: {
-            'content-type': 'application/json',
-            ...headers
-          }
-        })
-  },
-  put: (url, params = {}, headers = {}) =>
-    asyncFetch(url, {
-      method: 'PUT',
-      headers: {
-        'content-type': 'application/json',
-        ...headers
-      },
-      body: JSON.stringify(params)
-    }),
-  post: (url, params = {}) =>
-    asyncFetch(url, {
-      method: 'POST',
-      headers: {
-        'content-type': 'application/json'
-      },
-      body: JSON.stringify(params)
-    })
-}
-
-export function stringEnumerate([x0, ...rest]) {
-  if (!rest || !rest.length) {
-    return x0
-  }
-
-  return `${[x0, ...rest.slice(0, rest.length - 1)].join(', ')} and ${
-    rest[rest.length - 1]
-  }`
-}
-
-/** Allows await for a specified time interval */
-export const timeout = (ms) => new Promise((res) => setTimeout(res, ms))
-
-// thanks to https://stackoverflow.com/a/33379772/763705
-export function truncateOnWord(str, limit, end = '...') {
-  const trimmable =
-    '\u0009\u000A\u000B\u000C\u000D\u0020\u00A0\u1680\u180E\u2000\u2001\u2002\u2003\u2004\u2005\u2006\u2007\u2008\u2009\u200A\u202F\u205F\u2028\u2029\u3000\uFEFF'
-  const reg = new RegExp(`(?=[${trimmable}])`)
-  const words = str.split(reg)
-  let count = 0
-  const result = words
-    .filter((word) => {
-      count += word.length
-      return count <= limit
-    })
-    .join('')
-  return result + (result !== str ? end : '')
-}
-
-// thanks to https://stackoverflow.com/a/34695026/763705
-export function isValidURL(str) {
-  const a = document.createElement('a')
-  a.href = str
-  return a.host && a.host !== window.location.host
-}
-
-// thanks to https://stackoverflow.com/a/18650828/763705
-export function formatBytes(bytes, decimals = 2) {
-  if (bytes === 0) return '0 Bytes'
-  const k = 1024
-  const dm = decimals <= 0 ? 0 : decimals
-  const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB']
-  const i = Math.floor(Math.log(bytes) / Math.log(k))
-
-  return `${parseFloat((bytes / k ** i).toFixed(dm))} ${sizes[i]}`
-}
-
-/**
- * Format number to metric prefix (https://en.wikipedia.org/wiki/Metric_prefix)
- * numberFormatter(33000) = '33K'
- *
- * Thanks to https://stackoverflow.com/a/14994860/763705
- * @param {*} num
- * @param {*} digits
- */
-export function numberFormatter(num, digits = 0) {
-  const si = [
-    { value: 1, symbol: '' },
-    { value: 1e3, symbol: 'k' },
-    { value: 1e6, symbol: 'M' },
-    { value: 1e9, symbol: 'G' },
-    { value: 1e12, symbol: 'T' },
-    { value: 1e15, symbol: 'P' },
-    { value: 1e18, symbol: 'E' }
-  ]
-  const rx = /\.0+$|(\.[0-9]*[1-9])0+$/
-  let i
-  for (i = si.length - 1; i > 0; i -= 1) {
-    if (num >= si[i].value) {
-      break
-    }
-  }
-  return (num / si[i].value).toFixed(digits).replace(rx, '$1') + si[i].symbol
-}
```

Comment:
Was this copied from another project and now you've moved each of the functions this project actually uses into its own file?

Reply:
More or less yeah. They were copied from refine bio but I don't think I want to solve all of the problems the same way. I'd like to keep all of the helpers to be small descriptive bits of code that are reusable. Also it will be easier for me to check to see which files aren't being included so we can manage dead code better.